external-crates/move/crates/move-compiler/src/typing/translate.rs

```diff
@@ -1645,6 +1645,65 @@ fn binop(
             (Type_::bool(loc), operand_ty)
         }
 
+        Eq | Neq
+            if context
+                .env
+                .supports_feature(context.current_package(), FeatureGate::AutoborrowEq) =>
+        {
+            let lhs_type = core::ready_tvars(&context.subst, el.ty.clone());
+            let rhs_type = core::ready_tvars(&context.subst, er.ty.clone());
+            let (lhs_ref, lhs_inner, rhs_ref, rhs_inner) = match (lhs_type, rhs_type) {
+                (sp!(_, Type_::Ref(lhs_mut, lhs)), sp!(_, Type_::Ref(rhs_mut, rhs))) => {
+                    (Some(lhs_mut), *lhs, Some(rhs_mut), *rhs)
+                }
+                (sp!(_, Type_::Ref(lhs_mut, lhs)), rhs) => (Some(lhs_mut), *lhs, None, rhs),
+                (lhs, sp!(_, Type_::Ref(rhs_mut, rhs))) => (None, lhs, Some(rhs_mut), *rhs),
+                (lhs, rhs) => (None, lhs, None, rhs),
+            };
+            let ty = join(context, bop.loc, msg, lhs_inner.clone(), rhs_inner.clone());
+            context.add_single_type_constraint(loc, msg(), ty.clone());
+            let eq_ty = match (lhs_ref, rhs_ref) {
```

Comment:
nit, make a new variable in this scope, instead of having `mut` on the binop 
(something about having `mut` on a parameter for a translate function makes me uncomfortable lol)

Reply:
Yeah, agreed. Done, using a `return`.